mypy/typestate.py

```diff
@@ -0,0 +1,55 @@
+"""
+A shared state for all TypeInfo's that holds global cache and dependency information,
```

Comment:
Can you make it a bit more explicit that this is mutable global state (basically global variables)?

Grammar nit: TypeInfos, not TypeInfo's.